lib/pyfrc/physics/tankmodel.py

```diff
@@ -0,0 +1,316 @@
+                    
+                    self.drivetrain = tankmodel.TankModel.theory(motors.MOTOR_CFG_CIM,
+                                                                 robot_mass=90, gearing=10.71,
+                                                                 nmotors=2, x_wheelbase=2.0,
+                                                                 wheel_diameter=0.5)
+                    
+                def update_sim(self, hal_data, now, tm_diff):
+                    # TODO: get motor values from hal_data
+                    velocity, rotation = self.drivetrain.get_vector(l_motor, r_motor)
+                    self.physics_controller.drive(velocity, rotation, tm_diff)
+                    
+                    # optional: compute encoder
+                    # l_encoder = self.drivetrain.l_position * ENCODER_TICKS_PER_FT
+                    # r_encoder = self.drivetrain.r_position * ENCODER_TICKS_PER_FT
+    '''
+    
+    @classmethod
+    def theory(cls,
+               motor_config: MotorModelConfig,
+               robot_mass: float,
+               gearing: float,
+               nmotors: int = 1,
+               x_wheelbase: float = _kitbot_wheelbase,
+               robot_width: float = _kitbot_width,
+               robot_length: float = _kitbot_length,
+               wheel_diameter: float = 0.5,
+               vintercept: float = 1.3,
+               timestep_ms: int = 1):
+        r'''
+            Use this to create the drivetrain model when you haven't measured
+            ``kv`` and ``ka`` for your robot.
+            
+            :param motor_config:    Specifications for your motor
+            :param robot_mass:      Mass of the robot (in lb)
+            :param gearing:         Gear ratio .. so for a 10.74:1 ratio, you would pass 10.74
+            :param nmotors:         Number of motors per side
+            :param x_wheelbase:     Wheelbase of the robot (in feet)
+            :param robot_width:     Width of the robot (in feet)
+            :param robot_length:    Length of the robot (in feet)
+            :param wheel_diameter:  Diameter of the wheel (in feet)
+            :param vintercept:      The minimum voltage required to generate enough
+                                    torque to overcome steady-state friction (see the
+                                    paper for more details)
+            :param timestep_ms:     Model computation timestep (in ms)
+            
+            Computation of ``kv`` and ``ka`` are done as follows:
+            
+            .. math::
+            
+                velocity_{max} = \frac{\omega_{free} \cdot \pi \cdot d_{wheels} }{r_{gearing}}
+                
+                acceleration_{max} = \frac{2 \cdot n \cdot \tau_{stall} \cdot r_{gearing} }{d_{wheels} \cdot m_{robot}}
+                
+                k_{v} = \frac{V_{max}}{velocity_{max}}
+                
+                k_{a} = \frac{velocity_{max}}{acceleration_{max}}
```

Comment:
Rereading the paper, isn't _k<sub>a</sub>_ = _V_<sub>max</sub> / acceleration<sub>max</sub>?

Reply:
You are correct.